File: CzhERP.Application/Entity/StoStockOutItem.cs
namespace CzhERP.Application.Entity;

[Tenant("1300000000001")]
[SugarTable("Sto_StockOutItem", "出库单明细表")]
public partial class StoStockOutItem : EntityBase
{
    [Required]
    [SugarColumn(ColumnName = "StockOutId", ColumnDescription = "出库单ID")]
    public virtual long StockOutId { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialId", ColumnDescription = "物料ID")]
    public virtual long MaterialId { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialCode", ColumnDescription = "物料编码", Length = 50)]
    public virtual string MaterialCode { get; set; }

    [Required]
    [SugarColumn(ColumnName = "MaterialName", ColumnDescription = "物料名称", Length = 100)]
    public virtual string MaterialName { get; set; }

    [SugarColumn(ColumnName = "Spec", ColumnDescription = "规格型号", Length = 100)]
    public virtual string? Spec { get; set; }

    [Required]
    [SugarColumn(ColumnName = "Unit", ColumnDescription = "单位", Length = 20, DefaultValue = "个")]
    public virtual string Unit { get; set; } = "个";

    [Required]
    [SugarColumn(ColumnName = "Quantity", ColumnDescription = "出库数量")]
    public virtual decimal Quantity { get; set; }

    [SugarColumn(ColumnName = "UnitPrice", ColumnDescription = "单价", DefaultValue = "0")]
    public virtual decimal UnitPrice { get; set; } = 0;

    [SugarColumn(ColumnName = "Amount", ColumnDescription = "金额", DefaultValue = "0")]
    public virtual decimal Amount { get; set; } = 0;

    [SugarColumn(ColumnName = "LocationCode", ColumnDescription = "出库库位", Length = 50)]
    public virtual string? LocationCode { get; set; }

    [SugarColumn(ColumnName = "BatchNo", ColumnDescription = "批号", Length = 50)]
    public virtual string? BatchNo { get; set; }

    [SugarColumn(ColumnName = "SortOrder", ColumnDescription = "排序号", DefaultValue = "0")]
    public virtual int SortOrder { get; set; } = 0;

    [SugarColumn(ColumnName = "Remark", ColumnDescription = "备注", Length = 200)]
    public virtual string? Remark { get; set; }
}

File: Web/src/api/stock/stoStockOutItem.ts
import {useBaseApi} from '/@/api/base';

// 出库单明细表接口服务
export const useStoStockOutItemApi = () => {
	const baseApi = useBaseApi("stoStockOutItem");
	return {
		// 分页查询出库单明细表
		page: baseApi.page,
		// 查看出库单明细表详细
		detail: baseApi.detail,
		// 新增出库单明细表
		add: baseApi.add,
		// 更新出库单明细表
		update: baseApi.update,
		// 删除出库单明细表
		delete: baseApi.delete,
		// 批量删除出库单明细表
		batchDelete: baseApi.batchDelete,
		// 导出出库单明细表数据
		exportData: baseApi.exportData,
		// 导入出库单明细表数据
		importData: baseApi.importData,
		// 下载出库单明细表数据导入模板
		downloadTemplate: baseApi.downloadTemplate,
	}
}

// 出库单明细表实体
export interface StoStockOutItem {
	// 主键Id
	id: number;
	// 出库单ID
	stockOutId?: number;
	// 物料ID
	materialId?: number;
	// 物料编码
	materialCode?: string;
	// 物料名称
	materialName?: string;
	// 规格型号
	spec: string;
	// 单位
	unit?: string;
	// 出库数量
	quantity?: number;
	// 单价
	unitPrice?: number;
	// 金额
	amount?: number;
	// 出库库位
	locationCode: string;
	// 批号
	batchNo: string;
	// 排序号
	sortOrder?: number;
	// 备注
	remark: string;
	// 创建时间
	createTime: string;
	// 更新时间
	updateTime: string;
	// 创建者Id
	createUserId: number;
	// 创建者姓名
	createUserName: string;
	// 修改者Id
	updateUserId: number;
	// 修改者姓名
	updateUserName: string;
}